excel_import_export/models/xlsx_export.py

```diff
@@ -0,0 +1,271 @@
+    _logger.debug(
+        'Cannot import "openpyxl". Please make sure it is installed.')
+
+
+class XLSXExport(models.AbstractModel):
+    _name = 'xlsx.export'
+
+    @api.model
+    def get_eval_context(self, model, record, value):
+        eval_context = {'float_compare': float_compare,
+                        'time': time,
+                        'datetime': dt,
+                        'date': date,
+                        'value': value,
+                        'object': record,
+                        'model': self.env[model],
+                        'env': self.env,
+                        'context': self._context,
+                        }
+        return eval_context
+
+    @api.model
+    def _get_line_vals(self, record, line_field, fields):
+        """ Get values of this field from record set and return as dict of vals
+            - record: main object
+            - line_field: rows object, i.e., line_ids
+            - fields: fields in line_ids, i.e., partner_id.display_name
+        """
+        line_field, max_row = co.get_line_max(line_field)
+        line_field = line_field.replace('_CONT_', '')  # Remove _CONT_ if any
+        lines = record[line_field]
+        if max_row > 0 and len(lines) > max_row:
+            raise Exception(
+                _('Records in %s exceed max records allowed') % line_field)
+        vals = dict([(field, []) for field in fields])  # value and do_format
+        # Get field condition & aggre function
+        field_cond_dict = {}
+        aggre_func_dict = {}
+        field_format_dict = {}
+        format_cond_dict = {}
+        pair_fields = []  # I.e., ('debit${value and . or .}@{sum}', 'debit')
+        for field in fields:
+            temp_field, eval_cond = co.get_field_condition(field)
+            eval_cond = eval_cond or 'value or ""'
+            temp_field, field_format = co.get_field_format(temp_field)
+            temp_field, format_cond = co.get_field_format_cond(temp_field)
+            raw_field, aggre_func = co.get_field_aggregation(temp_field)
+            # Dict of all special conditions
+            field_cond_dict.update({field: eval_cond})
+            aggre_func_dict.update({field: aggre_func})
+            field_format_dict.update({field: field_format})
+            format_cond_dict.update({field: format_cond})
+            # --
+            pair_fields.append((field, raw_field))
+        for line in lines:
+            for field in pair_fields:  # (field, raw_field)
+                value = self._get_field_data(field[1], line)
+                eval_cond = field_cond_dict[field[0]]
+                eval_context = \
+                    self.get_eval_context(line._name, line, value)
+                if eval_cond:
+                    value = eval(eval_cond, eval_context)
+                # Format w/Cond takes priority
+                format_cond = format_cond_dict[field[0]]
+                format = self._eval_format_cond(line._name, line,
```

Comment:
`format` is a built-in method of python.
Then you are redefining it
What about using another variable name?

Reply:
Changed all 'format' to 'style' (also Style is better meaning in our case too). Thanks.